src/xenia/kernel/xam/xam_info.cc

```diff
@@ -200,8 +202,41 @@ dword_result_t XGetAVPack_entry() {
 }
 DECLARE_XAM_EXPORT1(XGetAVPack, kNone, kStub);
 
-uint32_t xeXGetGameRegion() { return 0xFFFFu; }
+// Going by the theory based on some unreasonable amount of
+// experimentation, I'll say that the top byte is the region here:
+// 00 is most likely USA, 01 Asia, 02 Europe, ???, FF = All regions?
+// And the lower byte is some sort of mask for sub-regions?
+// 0x0101 is the only one that displays the RIOT ACT title screen in
+// 4D5307DC (Crackdown), because this is only a copyrighted title in
+// Japan specifically, 0x01FF displays the standard CRACKDOWN title,
+// so 0x0101 is most likely Japan and 0x01<anything else> is probably
+// different sub regions of Asia?
+// Thus the logic behind these return values is:
+// 0xFFFF - Any region, all sub regions (basically region free)
+// 0x00FF - US, all sub regions
+// 0x0101 - Asia region, Japan sub region
+// 0x01FF - Asia region, all sub regions (Chinese/Hong Kong consoles?)
+// 0x02FF - Europe, all sub regions
+// 0x0300 through 0x06FF - Unknown (World, unused?)
+// 0x07FF - Devkit, according to text displayed on screen in DashLaunch.
+//          Probably not very useful for now? Not added.
+uint32_t xeXGetGameRegion() {
+  if (cvars::console_region == "us") {
+    return 0x00FFu;
+  } else if (cvars::console_region == "jp") {
+    return 0x0101u;
+  } else if (cvars::console_region == "asia") {
+    return 0x01FFu;
+  } else if (cvars::console_region == "eu") {
+    return 0x02FFu;
```

Comment:
These appear to correlate with xex region flags (listed in xex2_region_flags). So it likely goes like this:
0x00FF - North America
0x0101 - Japan
0x0102 - China
0x01FF - Asia
0x0201 - Australia
0x02FF - Europe

Reply:
Hi, thanks for this comment, but how did you come to that conclusion? I remember looking at the enums in `xex2_info.h` and not finding them particularly useful as to what value this should return, as it looks like this, which seems to be a 32-bit mask of some sort that I don't know the origin of:
```
enum xex2_region_flags : uint32_t {
  XEX_REGION_NTSCU = 0x000000FF,
  XEX_REGION_NTSCJ = 0x0000FF00,
  XEX_REGION_NTSCJ_JAPAN = 0x00000100,
  XEX_REGION_NTSCJ_CHINA = 0x00000200,
  XEX_REGION_PAL = 0x00FF0000,
  XEX_REGION_PAL_AU_NZ = 0x00010000,
  XEX_REGION_OTHER = 0xFF000000,
  XEX_REGION_ALL = 0xFFFFFFFF,
};
```
Like the comment in my code says, my values are based entirely on experimentation and in one (unused) case what DashLaunch displayed on screen for one value (`0x07FF`) when I typed it in on my RGH Xbox for the "Region spoofing" setting, but many of these values appeared to be wrong and/or empty.

I can of course add Australia as a preset value if you've checked this, and if it results in different game operation compared to just selecting "Europe". (That was the only reason I added Japan as a separate entry from what appeared to work as a generic NTSC-J "Asia" setting with any other `0x02XX` value.

Reply:
It lines up when you look at the same side-by-side - look at it this way, in xex2_region_flags, NTSC-U is byte 0, NTSC-J is byte 1 and PAL is byte 2 and the actual value within the byte is sub-region.

Reply:
I really don't know what you're talking about, sorry. The value returned from `xeXGetGameRegion` is definitely a 16-bit value, there's no third or fourth byte.
To be clear: I'm sure this is accurate for *XEX files*, but I still don't see any connection to the region value returned by the console and the region value set in the XEX file, and I'm pretty sure that *no* value from the XEX file should be used when checking for game region using this function.

Reply:
Yeah, you misunderstood me again. How about this, in xex2_region_flags, count octets from right to left. The first one is NTSC-U, the second one is NTSC-J, the third one is PAL. This correlates to values returned by xeXGetGameRegion where, in the bottom 8 bits, 0 - is NTSC-U, 1 is NTSC-J and 2 is PAL.

Reply:
Ah, sorry. I have no idea how octets work. It's not that I misunderstand, it's simply that I do not understand, like I said.
But I mean, it seems like you have a proper solution for this, and if you want to rework the code to use the proper math for it instead, you are of course free to do so, I have no idea how to do it.

Reply:
Octets is a group of 8 bits.
Take 0x12345678 number, for example. Since bits in a number are counted from right to left, 78 is the first octet, 56 is the second octet, 34 is the third octet, 12 is the fourth octet.
Do you see it now?

Reply:
No, sorry. I honestly have no idea. It's a pair of bits?
Please though, I'm glad that you found a proper solution for this, I really am.
But you really don't have to educate me on octets, it's fine. I fully admit that I have no idea, so please give me a bit of a break here.
I have no desire to understand octets, nor do I have any practical use for it. If something cannot be interpreted as a simple value or different bits set, that's way out of my league. I'm sorry if I somehow gave you the wrong impression, but I am not a software engineer by any means, I'm probably completely unsuitable for tasks like this, and I'm fine with that. I just do what I can figure out sometimes for fun, Anyone can close this pull request if it's unsuitable, or I can close it if asked to, it's really no big deal.